internal/util/tls/tls_cert.go

```diff
@@ -0,0 +1,48 @@
+package tls
+
+import (
+	"crypto/tls"
+	"fmt"
+	"os"
+)
+
+// ExtractClientCertificates extracts tls.Certificates from TLSClientConfig.
+// It returns an empty slice in case there was no client cert and/or client key provided.
+func ExtractClientCertificates(cert []byte, certFile string, key []byte, keyFile string) ([]tls.Certificate, error) {
```

Comment:
Is there an intention for this to return more than 1 cert? It doesn't look like from the current definition in here.

If that's the case I'd suggest to rename it to `ExtractClientCertificates` and return just a cert not a slice.

Reply:
I authored this function so I'll explain myself 😛 I decided to return a slice because it makes the caller code simpler as you can simply
```
clientCerts := ExtractClientCertificates(...)
tlsConfig.Certificates = append(tlsConfig.Certificates, clientCerts...) 
```
which will is valid for an empty slice as well. If that's not obvious enough, let's add a comment or change it as you suggested.

Reply:
As I wrote in another comment: I do reckon the fact that this works but... yeah, it's not obvious. And since there's no intention to return more than 1 then I don't see a reason to keep slice as a return value.

Reply:
The alternative then is to change the signature to `ExtractClientCertificate(...) (cert tls.Certificate, ok bool, err error)` as it's a valid case to get no certificate here.

Reply:
updated. add a `REVIEW` comment to receive your options.